optimism/src/keccak/mod.rs

```diff
@@ -131,23 +132,35 @@ impl<F: Field> KeccakCircuit<F> {
         });
     }
 
-    /// Pads the rows of the witnesses until reaching the domain size
-    pub(crate) fn pad_rows(&mut self) {
-        for step in [
-            Round(0),
-            Sponge(Absorb(First)),
-            Sponge(Absorb(Middle)),
-            Sponge(Absorb(Last)),
-            Sponge(Absorb(Only)),
-            Sponge(Squeeze),
-        ] {
-            let rows_left =
-                self.witness[&step].cols[0].capacity() - self.witness[&step].cols[0].len();
-            for _ in 0..rows_left {
-                self.push_row(step, &[F::zero(); ZKVM_KECCAK_COLS]);
+    fn pad(&mut self, step: Steps) -> bool {
```

Comment:
```suggestion
    /// Pads unused rows for a particular keccak step if needed. Returns true if padding was performed, false otherwise.
    fn pad(&mut self, step: Steps) -> bool {
```